[k8s][docs] Add feature cards for users and admins (#3582)
skypilot-org/skypilot: docs/source/reference/kubernetes/index.rst

Running on Kubernetes
=============================

SkyPilot tasks can be run on your private on-prem or cloud Kubernetes clusters.
The Kubernetes cluster gets added to the list of "clouds" in SkyPilot and SkyPilot
tasks can be submitted to your Kubernetes cluster just like any other cloud provider.

Why use SkyPilot on Kubernetes?
-------------------------------

.. tab-set::

    .. tab-item:: For AI Developers
        :sync: why-ai-devs-tab

        .. grid:: 2
            :gutter: 3

            .. grid-item-card::  ✅ Ease of Use
                :text-align: center

                No complex kubernetes manifests - write a simple SkyPilot YAML and run ``sky launch``.

> 

Author: I think that requires a little more work since we need to work on compressing our existing YAMLs (in terms of lines of code #3594) to get a good comparison against kubernetes manifests. E.g., [k8s vllm gemma manifest](https://cloud.google.com/kubernetes-engine/docs/tutorials/serve-gemma-gpu-vllm) is 65 lines, while SkyPilot gemma is 38 lines. We should do it as a part of the K8s blog post, marked it as a TODO for now.

            .. grid-item-card::  ☁️ Burst to the cloud
                :text-align: center

                Kubernetes cluster is full? SkyPilot seamlessly gets resources on the cloud to get your job running sooner.

            .. grid-item-card::  🖼 Run popular models on Kubernetes
                :text-align: center

                Train and serve `Llama-3 <https://skypilot.readthedocs.io/en/latest/gallery/llms/llama-3.html>`_, `Mixtral <https://skypilot.readthedocs.io/en/latest/gallery/llms/mixtral.html>`_, and more on your Kubernetes with ready-to-use recipes from the :ref:`AI gallery <ai-gallery>`.

    .. tab-item:: For Infrastructure Admins
        :sync: why-admins-tab

        .. grid:: 2
            :gutter: 3

            .. grid-item-card::  ☁️ Burst to the cloud
                :text-align: center

                Scale out to capacity across clouds and regions without requiring manual intervention.

            .. grid-item-card::  🚯️ Minimize resource wastage
                :text-align: center

>                 SkyPilot can automatically terminate idle pods to free up resources for other users.

Author: Rephrased as `SkyPilot can run with your custom pod scheduler and automatically terminate idle pods to free up resources for other users.` wdyt?

            .. grid-item-card::  👀 Observability
                :text-align: center

>                 Use your existing tools, such as the :ref:`Kubernetes Dashboard <kubernetes-observability>`, to monitor SkyPilot pods.

Reviewer: ```suggestion
                Works with your existing tools, such as the :ref:`Kubernetes Dashboard <kubernetes-observability>`.
```

* An existing Kubernetes cluster running Kubernetes v1.20 or later.
* A `Kubeconfig <https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/>`_ file containing access credentials and namespace to be used.

**Supported Kubernetes deployments:**

* Hosted Kubernetes services (EKS, GKE)
* On-prem clusters (Kubeadm, Rancher, K3s)

> * Local development clusters (KinD, minikube)

Reviewer: How about we rename `Submitting SkyPilot tasks to Kubernetes Clusters` to `Getting Started` and make it a subpage of `Run On Kubernetes` page?

  For isolation, you can create separate Kubernetes namespaces and set them in the kubeconfig distributed to users. SkyPilot will use the namespace set in the kubeconfig for running all tasks.

* **How do I view the pods created by SkyPilot on my Kubernetes cluster?**

>   You can use your existing observability tools to filter resources with the label :code:`parent=skypilot`. As an example, follow the instructions :ref:`here <kubernetes-observability>` to deploy the Kubernetes Dashboard on your cluster.

Reviewer: ```suggestion
  You can use your existing observability tools to filter resources with the label :code:`parent=skypilot` (:code:`k get pods -l 'parent=skypilot'`). As an example, follow the instructions :ref:`here <kubernetes-observability>` to deploy the Kubernetes Dashboard on your cluster.
```
